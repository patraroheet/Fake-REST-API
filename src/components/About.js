import React, { Component } from "react";
import { Container, Button } from "reactstrap";
class About extends Component {
  goBack = () => {
    this.props.history.goBack();
  };

  render() {
    return (
      <Container>
        <h1>About Component</h1>
        <Button color="primary" onClick={this.goBack}>
          Back
        </Button>
      </Container>
    );
  }
}

export default About;
