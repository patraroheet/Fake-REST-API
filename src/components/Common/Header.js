import React from "react";
import {
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  NavItem
} from "reactstrap";
import { NavLink } from "react-router-dom";

export default class Header extends React.Component {
  constructor(props) {
    super(props);

    this.toggle = this.toggle.bind(this);
    this.state = {
      isOpen: false
    };
  }

  toggle() {
    this.setState({
      isOpen: !this.state.isOpen
    });
  }

  render() {
    return (
      <div>
        <Navbar color="light" light expand="md">
          <NavbarBrand href="/">Demo</NavbarBrand>
          <NavbarToggler onClick={this.toggle} />
          <Collapse isOpen={this.state.isOpen} navbar>
            <Nav className="ml-auto" navbar>
              <NavItem>
                <NavLink className={"nav-link"} exact to="/">
                  Home
                </NavLink>
              </NavItem>
              <NavItem>
                <NavLink className={"nav-link"} to="/posts">
                  Posts
                </NavLink>
              </NavItem>
              <NavItem>
                <NavLink className={"nav-link"} to="/about">
                  About
                </NavLink>
              </NavItem>
            </Nav>
          </Collapse>
        </Navbar>
      </div>
    );
  }
}
