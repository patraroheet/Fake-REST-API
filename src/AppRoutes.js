import React from "react";
import {Switch, Route} from "react-router-dom";
import Home from "./components/Home";
import NotFound from "./components/Common/NotFound";
import Posts from "./components/Posts";
import About from "./components/About";
import Auth from "./components/Authentication";
import PrivateRoute from "./core/PrivateRoute";

export default function AppRoutes() {
    return (
        <Switch>
            <Route path="/" exact component={Home}/>
            <PrivateRoute path="/posts" component={Posts}/>
            <Route path="/about" component={About}/>
            <Route path="/auth" component={Auth}/>
            <Route component={NotFound}/>
        </Switch>
    );
}
