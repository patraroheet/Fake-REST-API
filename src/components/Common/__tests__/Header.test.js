import React from "react";
import { mount, shallow } from "enzyme";
import Header from "../Header";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import AppRoutes from "../../../AppRoutes";
import About from "../../About";
import TestRenderer from "react-test-renderer";

describe("header component testing", () => {
  test("shall have correct snapShot", () => {
    const component = TestRenderer.create(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("shall render  without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <Header />
        <AppRoutes />
      </MemoryRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  test("shall render  about component on about link click", () => {
    const wrapper = mount(
      <MemoryRouter>
        <Header />
        <AppRoutes />
      </MemoryRouter>
    );
    // console.log(wrapper.debug());
    wrapper
      .find(Header)
      .find("[href='/about']")
      .simulate("click", { button: 0 });
    expect(wrapper.find(About)).toHaveLength(1);
  });
});
