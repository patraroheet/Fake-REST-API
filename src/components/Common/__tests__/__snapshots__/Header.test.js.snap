// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component testing shall have correct snapShot 1`] = `
<div>
  <nav
    className="navbar navbar-expand-md navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      Demo
    </a>
    <button
      className="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      style={Object {}}
    >
      <ul
        className="ml-auto navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/posts"
            onClick={[Function]}
          >
            Posts
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
